kafka/4.2/modules/ROOT/pages/kafka-connector-reference.adoc

```diff
@@ -0,0 +1,1558 @@
+| TLS Configuration a| <<Tls>> |  Protocol to use for communication. Valid values are HTTP (default) and HTTPS. When using HTTPS, the HTTP communication is secured using TLS or SSL. If HTTPS was configured as the protocol, then the user needs to configure at least the keystore in the `tls:context` child element of this `listener-config`. |  |
+| Reconnection a| <<Reconnection>> |  When the application is deployed, a connectivity test is performed on all connectors. If set to true, deployment fails if the test doesn't pass after exhausting the associated reconnection strategy |  |
+|===
+
+[[producer-sasl-scram]]
+===== Producer SASL SCRAM Connection
+
+====== Parameters
+[%header,cols="20s,20a,35a,20a,5a"]
+|===
+| Name | Type | Description | Default Value | Required
+| Username | String | The username with which to login. |  | x
+| Password | String | The password with which to login. |  | x
+| EncryptionType | Enumeration, one of:
+
+** SHA256
+** SHA512 | The encryption algorithm used by SCRAM. Only acceptable values are SHA-256 and SHA-512. |  | x
+| Bootstrap server URLs a| Array of String |  The list of servers to bootstrap the connection with the Kafka cluster. This can be a partial list of the available servers. |  | x
+| Batch size a| Number |  The producer attempts to batch records together into fewer requests whenever multiple records are sent to the same partition. This helps performance on both the client and the server. This configuration controls the default batch size in bytes. No attempt is made to batch records larger than this size. Requests sent to brokers will contain multiple batches, one for each partition with the data that is available to send. A small batch size makes batching less common and can reduce throughput (a batch size of zero disables batching entirely). A very large batch size can result in more wasteful use of memory as a buffer of the specified batch size is always allocated in anticipation of additional records. |  `16` |
+| The batch size unit of measure. a| Enumeration, one of:
+
+** BYTE
+** KB
+** MB
+** GB |  The unit of measure for the batch size scalar |  `KB` |
+| Buffer size a| Number |  The total bytes of memory the producer can use to buffer records waiting to be sent to the server. If records are sent faster than they can be delivered to the server, the producer will block for `max.block.ms` after which it throws an exception. This setting should generally correspond to the total memory the producer will use, but is not exact because not all memory the producer uses is used for buffering. Some additional memory is used for compression (if compression is enabled) as well as for maintaining in-flight requests. The default value in the Apache Kafka documentation is 33554432 (32MB), but this should be capped to align with expected values for Mule instances in CloudHub (v0.1 core) |  1 |
```

Comment:
```suggestion
| Buffer size a| Number |  The total bytes of memory the producer can use to buffer records waiting to be sent to the server. If records are sent faster than they can be delivered to the server, the producer blocks for `max.block.ms`, after which it throws an exception. This setting should generally correspond to the total memory the producer will use, but is not exact because the memory used by the producer is not all used for buffering. Some additional memory is used for compression (if compression is enabled), as well as for maintaining in-flight requests. The default value in the Apache Kafka documentation is `33554432` (32MB), but this should be capped to align with expected values for Mule instances in CloudHub (v0.1 core). |  1 |
```